docs/concepts/troubleshooting-installed-packages.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Troubleshooting Installed Packages
+description: How to find which package source was used for individual packages
+author: JonDouglas
+ms.author: jodou
+ms.date: 03/26/2021
+ms.topic: conceptual
+---
+
+# Troubleshooting Installed Packages
+
+Sometimes you might want to validate which source a specific package was installed from. Here are some ways you can check.
+
+## `.nupkg.metadata` file in global packages folder
+
+When a package is extracted into the *global-packages* folder, a file `.nupkg.metadata` is written. Starting from NuGet 5.9.0, NuGet will add the package source. See below to map NuGet versions to Visual Studio or .NET SDK versions. For example:
+
+```json
+{
+  "version": 2,
+  "contentHash": "bw3R9q8cVNhWXNpnvWb0OGP4HadS4zvClq+T1zf7AF+tLY1haZ2AvbHidQekf4PDv1T40c6brZeT/V0IBq7cEQ==",
+  "source": "https://api.nuget.org/v3/index.json"
+}
+```
+
+> [!Note]
+> If your *global-packages* folder has packages extracted before you upgraded to a newer version of tools that has NuGet 5.9.0, the `.nupkg.metadata` file will be version 1 and will not contain the package source. You can [clear your *global-packages* folder](../consume-packages/managing-the-global-packages-and-cache-folders.md#clearing-local-folders) to ensure all packages will contain the package source.
+
+> [!Tip]
+> NuGet writes the `.nupkg.metadata` file to the *global-packages* folder only. Projects using `packages.config` use a solution packages folder, which does not create a `.nupkg.metadata` file. Unless `-NoCache` or `--no-cache` was used on the command line, the package will be in both the *global-packages*, with a `.nupkg.metadata` file, and solution packages folders, without the `.nupkg.metadata` file.
```

Comment:
Did you verify this?

Reply:
I have verified the 5.9.0 item. Could be simplified to 16.9 if wanted.

Reply:
My question is more about no-cache. 

no-cache should affect http cache only.

Reply:
You're right. I read the code not too long ago, but also not very recently, and my memory was that `NoCache` skipped the global packages folder. But that's wrong.